File: app/src/main/java/buggy/plandroid/com/plandroidbugs/room/UserDao.java
package buggy.plandroid.com.plandroidbugs.room;

import android.arch.persistence.room.*;

import java.util.Collection;
import java.util.List;

import io.reactivex.Flowable;

@Dao
public interface UserDao {
    @Query("SELECT * FROM users")
    Flowable<List<UserEntity>> getAll();

    @Query("SELECT * FROM users WHERE uid IN (:userIds)")
    List<UserEntity> loadAllByIds(int[] userIds);

    @Query("SELECT * FROM users WHERE first_name LIKE :first AND "
           + "last_name LIKE :last LIMIT 1")
    UserEntity findByName(String first, String last);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAll(Collection<UserEntity> userEntitys);

    @Update
    void update(UserEntity... userEntity);

    @Delete
    void delete(UserEntity userEntity);
}

File: app/src/main/java/buggy/plandroid/com/plandroidbugs/room/UserDao.kt
package buggy.plandroid.com.plandroidbugs.room

import androidx.room.*
import io.reactivex.Flowable


@Dao
interface UserDao {

    @Query("SELECT * FROM users")
    fun all(): Flowable<List<UserEntity>>

    @Query("SELECT * FROM users WHERE uid IN (:userIds)")
    fun loadAllByIds(userIds: IntArray?): List<UserEntity>

    @Query("SELECT * FROM users WHERE first_name LIKE :first AND last_name LIKE :last LIMIT 1")
    fun findByName(first: String?, last: String?): UserEntity?

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAll(userEntityies: Collection<UserEntity>)

    @Update
    fun update(vararg userEntity: UserEntity)

    @Delete
    fun delete(userEntity: UserEntity)
}